### PR TITLE
coordinate rotations

### PR DESCRIPTION
**Pull request by lheagy**

### CODE REVIEW

**Comment by lheagy** [comment]
Do you think the name coordutils makes sense? @rowanc1 @sgkang @grosenkj

File: SimPEG/Utils/coordutils.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import numpy as np
+
+def crossProd(v0,v1):
```
Comment (rowanc1) [inline SimPEG/Utils/coordutils.py]: np.cross?

File: SimPEG/Utils/coordutils.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    assert len(v1) == 3, "Length of n1 should be 3"
+
+    # ensure both are true normals
+    n0 = v0*1./np.linalg.norm(v0)
+    n1 = v1*1./np.linalg.norm(v1)
+
+    n0dotn1 = n0.dot(n1) 
+
+    # define the rotation axis, which is the cross product of the two vectors
+    rotAx = crossProd(n0,n1)
```
Comment (rowanc1) [inline SimPEG/Utils/coordutils.py]: `np.cross(n0,n1)`

File: SimPEG/Utils/coordutils.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        :rtype: numpy.array, 3x3
+        :return: rotation matrix which rotates the frame so that n0 is aligned with n1
+    """
+
+    R = rotationMatrixFromNormals(n0, n1)
+
+    assert XYZ.shape[1] == 3, "Grid XYZ should be 3 wide"
+    assert len(x0) == 3, "x0 should have length 3"
+
+    return (XYZ - x0).dot(R.T) + x0
```
Comment (rowanc1) [inline SimPEG/Utils/coordutils.py]: XYZ - x0 if your XYZ is long (n*3) this wont work?

Comment (lheagy) [inline SimPEG/Utils/coordutils.py]: do this on each column

**Pull request merged**